zebrad/src/components/mempool/tests/prop.rs

```diff
@@ -0,0 +1,181 @@
+            .enable_all()
+            .build()
+            .expect("Failed to create Tokio runtime");
+        let _guard = runtime.enter();
+
+        runtime.block_on(async move {
+            let (
+                mut mempool,
+                mut peer_set,
+                mut state_service,
+                mut tx_verifier,
+                mut recent_syncs,
+                _chain_tip_sender,
+            ) = setup(network);
+
+            time::pause();
+
+            mempool.enable(&mut recent_syncs).await;
+
+            // Insert a dummy transaction.
+            mempool
+                .storage()
+                .insert(transaction)
+                .expect("Inserting a transaction should succeed");
+
+            // The first call to `poll_ready` shouldn't clear the storage yet.
+            dummy_service_call(&mut mempool).await;
+
+            prop_assert_eq!(mempool.storage().tx_ids().len(), 1);
+
+            // Simulate the synchronizer catching up to the network chain tip.
+            for _ in 0..10 {
+                recent_syncs.push_extend_tips_length(50);
+            }
+
+            // This time a call to `poll_ready` should clear the storage.
+            dummy_service_call(&mut mempool).await;
+
+            // Enable the mempool again so the storage can be accessed.
+            mempool.enable(&mut recent_syncs).await;
+
+            prop_assert!(mempool.storage().tx_ids().is_empty());
+
+            peer_set.expect_no_requests().await?;
+            state_service.expect_no_requests().await?;
+            tx_verifier.expect_no_requests().await?;
+
+            Ok(())
+        })?;
+    }
+}
+
+/// Create a new [`Mempool`] instance using mocked services.
+fn setup(
+    network: Network,
+) -> (
+    Mempool,
+    MockPeerSet,
+    MockState,
+    MockTxVerifier,
+    RecentSyncLengths,
+    ChainTipSender,
+) {
+    let peer_set = MockService::build().for_prop_tests();
+    let state_service = MockService::build().for_prop_tests();
+    let tx_verifier = MockService::build().for_prop_tests();
+
+    let (sync_status, recent_syncs) = SyncStatus::new();
+    let (chain_tip_sender, latest_chain_tip, chain_tip_change) = ChainTipSender::new(None, network);
+
+    let mempool = Mempool::new(
+        network,
+        Buffer::new(BoxService::new(peer_set.clone()), 1),
+        Buffer::new(BoxService::new(state_service.clone()), 1),
+        Buffer::new(BoxService::new(tx_verifier.clone()), 1),
+        sync_status,
+        latest_chain_tip,
+        chain_tip_change,
+    );
+
+    (
+        mempool,
+        peer_set,
+        state_service,
+        tx_verifier,
+        recent_syncs,
+        chain_tip_sender,
+    )
+}
+
+/// Perform a dummy service call so that `poll_ready` is called.
+async fn dummy_service_call(mempool: &mut Mempool) {
+    mempool
+        .oneshot(Request::Queue(vec![]))
+        .await
+        .expect("Queuing no transactions shouldn't fail");
+}
```

Comment:
Hmm, I'm wondering if it would make sense to add this as a method to `Mempool`, so we can do something like `mempool.dummy_call().await` :thinking: 

Given the other test-specific methods in `Mempool`, I was also wondering if it would make sense to move them into a separate `#[cfg(test)] impl Mempool` inside the `tests` module. Just throwing an idea :sweat_smile:

Reply:
I like the idea!

Reply:
Done in [7ab4f4a](https://github.com/ZcashFoundation/zebra/pull/2815/commits/7ab4f4a302798d3a9f753dbf5d3cc61be12d6dc0), [c8d6917](https://github.com/ZcashFoundation/zebra/pull/2815/commits/c8d6917270507c5e74bac866144fe95aa28b3827) and [d15f486](https://github.com/ZcashFoundation/zebra/pull/2815/commits/d15f48694e8958a515899bf5c14a6cd1d45c8e95).